core/ledger/confighistory/mgr_test.go

```diff
@@ -257,6 +260,198 @@ func TestWithImplicitColls(t *testing.T) {
 
 }
 
+type testEnvForSnapshot struct {
+	mgr             *mgr
+	retriever       *Retriever
+	testSnapshotDir string
+	cleanup         func()
+}
+
+func newTestEnvForSnapshot(t *testing.T) *testEnvForSnapshot {
+	dbPath, err := ioutil.TempDir("", "confighistory")
+	if err != nil {
+		t.Fatalf("Failed to create config history directory: %s", err)
+	}
+	p, err := newDBProvider(dbPath)
+	if err != nil {
+		os.RemoveAll(dbPath)
+		t.Fatalf("Failed to create new leveldb provider: %s", err)
+	}
+	mgr := &mgr{dbProvider: p}
+	retriever := mgr.GetRetriever("ledger1", nil)
+
+	testSnapshotDir, err := ioutil.TempDir("", "confighistorysnapshot")
+	if err != nil {
+		os.RemoveAll(dbPath)
+		t.Fatalf("Failed to create config history snapshot directory: %s", err)
+	}
+	return &testEnvForSnapshot{
+		mgr:             mgr,
+		retriever:       retriever,
+		testSnapshotDir: testSnapshotDir,
+		cleanup: func() {
+			os.RemoveAll(dbPath)
+			os.RemoveAll(testSnapshotDir)
+		},
+	}
+}
+
+func TestExportConfigHistory(t *testing.T) {
+	env := newTestEnvForSnapshot(t)
+	defer env.cleanup()
+	// config history database is empty
+	fileHashes, err := env.retriever.ExportConfigHistory(env.testSnapshotDir, sha256.New())
+	require.NoError(t, err)
+	verifyExportedConfigHistory(t, env.testSnapshotDir, fileHashes, nil)
+	os.Remove(path.Join(env.testSnapshotDir, snapshotDataFileName))
+	os.Remove(path.Join(env.testSnapshotDir, snapshotMetadataFileName))
+
+	// config history database has 3 chaincodes each with 1 collection config entry in the
+	// collectionConfigNamespace
+	dbHandle := env.mgr.dbProvider.getDB("ledger1")
+	cc1collConfigPackage := testutilCreateCollConfigPkg([]string{"Explicit-coll-1", "Explicit-coll-2"})
+	cc2collConfigPackage := testutilCreateCollConfigPkg([]string{"Explicit-coll-1", "Explicit-coll-2"})
+	cc3collConfigPackage := testutilCreateCollConfigPkg([]string{"Explicit-coll-1", "Explicit-coll-2"})
+	batch, err := prepareDBBatch(
+		map[string]*peer.CollectionConfigPackage{
+			"chaincode1": cc1collConfigPackage,
+			"chaincode2": cc2collConfigPackage,
+			"chaincode3": cc3collConfigPackage,
+		},
+		50,
+	)
```

Comment:
I typically prefer to add some variation in the data (the content of collconfig and the committing block num) for different chaincodes. This makes it a better test - that can catch any bug if we end up mixing the data across chaincodes.

Reply:
Agree. Done.